app/script/view_model/content/MessageListViewModel.js

```diff
@@ -80,41 +80,43 @@ z.viewModel.content.MessageListViewModel = class MessageListViewModel {
 
     this.recalculate_timeout = undefined;
 
-    // Should we scroll to bottom when new message comes in
-    this.should_scroll_to_bottom = true;
-
-    // Check if the message container is to small and then pull new events
-    this.on_mouse_wheel = _.throttle(event => {
-      const is_not_scrollable = !$(event.currentTarget).isScrollable();
-      const is_scrolling_up = event.deltaY > 0;
-
-      if (is_not_scrollable && is_scrolling_up) {
-        return this._pull_messages();
+    this.onMouseWheel = _.throttle((data, event) => {
+      const element = $(event.currentTarget);
+      if (element.isScrollable()) {
+        // if the element is scrollable, the scroll event will take the relay
+        return true;
       }
-    }, 200);
-
-    this.on_scroll = _.throttle((data, event) => {
-      if (this.capture_scrolling_event) {
-        const element = $(event.currentTarget);
-
-        // On some HDPI screen scrollTop returns a floating point number instead of an integer
-        // https://github.com/jquery/api.jquery.com/issues/608
-        const scroll_position = Math.ceil(element.scrollTop());
-        const scrollEnd = element.scrollEnd();
+      const isScrollingUp = event.deltaY > 0;
+      const loadExtraMessagePromise = isScrollingUp ? this._loadPrecedingMessages() : this._loadFollowingMessages();
 
-        if (scroll_position === 0) {
-          this._pull_messages();
+      loadExtraMessagePromise.then(() => {
+        const antiscroll = $('.message-list').data('antiscroll');
+        if (antiscroll) {
+          antiscroll.rebuild();
         }
+      });
 
-        if (scroll_position >= scrollEnd) {
-          if (this._conversationHasExtraMessages(this.conversation())) {
-            this._push_messages();
-          }
+      return true;
+    }, 50);
 
-          this._mark_conversation_as_read_on_focus(this.conversation());
-        }
-
-        this.should_scroll_to_bottom = scroll_position > scrollEnd - z.config.SCROLL_TO_LAST_MESSAGE_THRESHOLD;
```

Comment:
This has been moved to a getter that returns true when we need to stick to the bottom